src/clarity-icons/clarity-icons-api.ts

```diff
@@ -17,64 +18,60 @@ export class ClarityIconsApi {
     }
 
     static get instance(): ClarityIconsApi {
-
         if (!ClarityIconsApi.singleInstance) {
-
             ClarityIconsApi.singleInstance = new ClarityIconsApi();
-
         }
 
         return ClarityIconsApi.singleInstance;
-
     }
 
     private validateName(name: string): boolean {
-
         if (name.length === 0) {
-
             throw new Error("Shape name or alias must be a non-empty string!");
         }
 
-
         if (/\s/.test(name)) {
-
             throw new Error("Shape name or alias must not contain any whitespace characters!");
-
         }
 
         return true;
-
     }
 
-    private validateTemplate(template: string): boolean {
-
-
-        if (!template.startsWith("<svg") || !template.endsWith("</svg>")) {
-
-            throw new Error("Template must be SVG markup!");
-
-        }
-
-        return true;
-
+    private sanitizeTemplate(template: string): string {
+        const allowedTags = [
+            "img", "div", "span", "svg", "animate", "animateMotion", "animateTransform",
+            "circle", "clipPath", "defs", "desc", "ellipse", "feBlend", "feColorMatrix",
+            "feComponentTransfer", "feComposite", "feConvolveMatrix", "feDiffuseLighting",
+            "feDisplacementMap", "feDistantLight", "feDropShadow", "feFlood", "feFuncA",
+            "feFuncB", "feFuncG", "feFuncR", "feGaussianBlur", "feImage", "feMerge", "feMergeNode",
+            "feMorphology", "feOffset", "fePointLight", "feSpecularLighting", "feSpotLight",
+            "feTile", "feTurbulence", "filter", "g", "line", "linearGradient", "marker",
+            "mask", "mpath", "path", "pattern", "polygon", "polyline", "radialGradient", "rect",
+            "stop", "symbol", "text", "textPath", "title", "use", "view"
+        ];
+
+        const sanitizeOptions = {
+            SAFE_FOR_TEMPLATES: true,
+            FORBID_ATTR: ["style"],
```

Comment:
Yes, sadly. There's way too many harmful things that can be done with style attrs. The recommendation is to load your SVG in an img tag if it requires styles. Or move those styles up into your own CSS files.